### ISSUE TITLE
drachtio-server with fsmrf (conference)

### ISSUE THREAD

**Issue by andrewvmail** [davehorton/drachtio-server#68]
Hi Dave, 

Happy new year for starter, hope you had a good new year.

Okay so I'm trying to develop a conferencing solution using fsmrf and I'm getting a crash when endpoint.destroy() is called.
crash log here: https://gist.github.com/andrewvmail/82139730fbd4b6593af947c0278e6fdc 

I'm on the v0.7.3-12-gfad3d4cc9 (nat-handling-branch)

The crash happens during development. The setup is as follows:
- drachtio-server running on a mac host computer
- drachtio freeswitch docker container running on the mac docker virtual machine (192.168.99.100)
- I made a slight modification to mrf.xml so that sip-ip, tp-ip, ext-sip-ip, ext-rtp-ip is binded to the above vm ip instead of the containers ip address

I can create conference and connect to it, but dialog.destroy() crash the drachtio server.

**Comment by davehorton** [comment]
can you provide me with a backtrace (stack trace) from that crash?

**Comment by andrewvmail** [comment]
@davehorton  I forgot to grab the back trace for that one, here is a new crash with it https://gist.github.com/andrewvmail/bf6372003c2c0970783df9cd23ade2d2

**Comment by davehorton** [comment]
What's very strange here, and probably related to the problem, is that the INVITE for the dialog is sent over tcp, but then the BYE coming back (from the freeswitch) is over UDP.

Assuming this INVITE-BYE dialog is between the drachtio server and the freeswitch, can you get a debug log from the freeswitch?  I'd like to figure out why it is sending the BYE over UDP when the dialog was created using TCP.

Also, was it your intent to use SIP over TCP for the connection to the media server?

**Comment by andrewvmail** [comment]
Hmm the Invite should be coming from websocket? let me dig that Freeswitch log now.

**Comment by andrewvmail** [comment]
Okay here is the Freeswitch log  https://gist.github.com/andrewvmail/65b6e3c1eb71a8f9fffc80804e5d49bf

Oh and I didnt know it was using TCP for the connection to the media server...  It should be the default. The client is using websockets to connect to Drachtio

**Comment by davehorton** [comment]
can you redo the freeswitch trace, and in addition to turning up sofia logging as you have done, also do this:
```
fsctl loglevel debug
```
so I can see dialplan logging and everything else

**Comment by andrewvmail** [comment]
Okay more this is after turning on ``` fsctl loglevel debug ```
https://gist.github.com/andrewvmail/17da7b823ca6608698581f29a384ea40

**Comment by davehorton** [comment]
It looks like I have some issues related to the use of TCP, and freeswitch is also complicating things by responding to an INVITE sent over TCP with a 200 ok including a Contact header with transport UDP.  There are a few things to look into, and it may be several days before I can really drill into this.  In the meantime, can you test this simple change to see if it is a workaround by causing the traffic to freeswitch to be UDP from the start.

In the config, change this:
```
<contact>sip:0.0.0.0:5040;transport=tcp,udp</contact>
```
to this:
```
<contact>sip:0.0.0.0:5040;transport=udp,tcp</contact>
```
and let me know if that works, and if it fixes the crash.

**Comment by andrewvmail** [comment]
I didn't realize i mixed up the default order from the example here https://github.com/davehorton/drachtio-server/blob/master/drachtio.conf.xml
I think i typed it in by hand accidentally switch ```udp,tcp``` to ```tcp,udp``` didn't realize the order matters. But even though i change the order it still crashing and sending out the invite to freeswitch through TCP first for some reason...

``` <contact>sip:0.0.0.0:5040;transport=udp,tcp</contact> ```  doesnt work
but
``` <contact>sip:0.0.0.0:5040;transport=udp</contact> ``` does work

I will leave this open for you to close since the underlying bug is still there?

Thanks alot Dave!

PS: fsmrf is awesome it beats working with freeswitch directly! and i found that I think your right a while back ago you mentioned https://github.com/davehorton/drachtio-srf/issues/10#issuecomment-391189326 is quicker than webrtc + coturn for some reason ...

**Comment by davehorton** [comment]
Yes, let's leave this open.  Thx for the comments, its great to know it is being of use.  I'm coming back to spending a bit more time on fsmrf in the next few weeks than I have for a while, so hopefully I can get to the bottom of this and make some other improvements!
